Objects/structseq.c

```diff
@@ -365,8 +366,35 @@ structseq_reduce(PyStructSequence* self, PyObject *Py_UNUSED(ignored))
     return NULL;
 }
 
+static PyObject *
+structseq_asdict(PyStructSequence* self, PyObject *Py_UNUSED(ignored))
+{
+    PyObject* dict = NULL;
+    Py_ssize_t n_visible_fields, n_unnamed_fields, i;
+
+    n_visible_fields = VISIBLE_SIZE(self);
+    n_unnamed_fields = UNNAMED_FIELDS(self);
+
+    dict = PyDict_New();
+    if (!dict)
+        return NULL;
+
+    for (i = 0; i < n_visible_fields; i++) {
+        const char *n = Py_TYPE(self)->tp_members[i-n_unnamed_fields].name;
+        if (PyDict_SetItemString(dict, n, self->ob_item[i]) < 0)
+            goto error;
+    }
+
+    return dict;
+
+error:
+    Py_XDECREF(dict);
```

Comment:
`error` is only reachable after `dict = PyDict_New();` succeeded. 

So, you can use `Py_DECREF`, `dict` cannot be `NULL` at that point.

Reply:
Thanks for catching it, pushed the proposed change.